src/coprocessor/codec/data_type/vector.rs

```diff
@@ -15,26 +15,22 @@ use std::convert::{TryFrom, TryInto};
 
 use cop_datatype::{EvalType, FieldTypeAccessor, FieldTypeTp};
 
+use super::*;
 use crate::coprocessor::codec::datum;
 use crate::coprocessor::codec::mysql::Tz;
 use crate::coprocessor::codec::{Error, Result};
 use crate::util::codec::{bytes, number};
 
-// TODO: Move these type alias and re-exports into cop_datatype.
-// These types are ensured to be cheap to move. However clone can be expensive.
-pub type Int = i64;
-pub type Real = f64;
-pub type Bytes = Vec<u8>;
-pub use crate::coprocessor::codec::mysql::{Decimal, Duration, Json, Time as DateTime};
-
-/// An array of datums in the same data type and is column oriented.
+/// A vector value container, a.k.a. column, for all concrete eval types.
 ///
-/// Stores datums of multiple rows of one column.
+/// The inner concrete value is immutable. However it is allowed to push and remove values from
+/// this vector container.
 #[derive(Debug, PartialEq)]
 pub enum VectorValue {
     Int(Vec<Option<Int>>),
     Real(Vec<Option<Real>>),
     Decimal(Vec<Option<Decimal>>),
+    // TODO: We need to improve its performance, i.e. store strings in adjacent memory places
```

Comment:
Here would we use both `u64` and `i64` as `Int`?

Reply:
@AndreMouche Yes, both `u64` and `i64` are stored in `i64`, aliased to `Int`, in the same way as what we did now.